File: backend/app/routers/skills.py
from fastapi import APIRouter

from app.models.skill import DepartmentSummary, SkillSummary
from app.services import graph_service

router = APIRouter(tags=["skills"])


@router.get("/skills", response_model=list[SkillSummary])
def list_skills():
    return graph_service.list_skills()


@router.get("/departments", response_model=list[DepartmentSummary])
def list_departments():
    return graph_service.list_departments()

File: frontend/src/api/skills.ts
import { apiGet } from "./client";
import type { DepartmentSummary, SkillSummary } from "../types";

export function listSkills() {
  return apiGet<SkillSummary[]>("/skills");
}

export function listDepartments() {
  return apiGet<DepartmentSummary[]>("/departments");
}
